reference/events.rst

```diff
@@ -0,0 +1,244 @@
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ResponseListener`             0
+:class:`Symfony\\Bundle\\SecurityBundle\\EventListener\\ResponseListener`            0
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ProfilerListener`             -100
+:class:`Symfony\\Bundle\\FrameworkBundle\\EventListener\\TestSessionListener`        -128
+:class:`Symfony\\Bundle\\WebProfilerBundle\\EventListener\\WebDebugToolbarListener`  -128
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\StreamedResponseListener`     -1024
+===================================================================================  ========
+
+``kernel.terminate``
+~~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\PostResponseEvent`
+
+The purpose of this event is to perform "heavier" tasks after the response
+was already served to the client.
+
+.. seealso::
+
+    Read more on the :ref:`kernel.terminate event <component-http-kernel-kernel-terminate>`.
+
+This is the built-in Symfony listener related to this event:
+
+=========================================================================  ========
+Listener Class Name                                                        Priority
+=========================================================================  ========
+`EmailSenderListener`_                                                     0
+=========================================================================  ========
+
+
+.. _kernel-kernel.exception:
+
+``kernel.exception``
+~~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\GetResponseForExceptionEvent`
+
+The FrameworkBundle registers an
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ExceptionListener` that
+forwards the ``Request`` to a given Controller (the value of the
+``exception_listener.controller`` parameter -- must be in the
+``class::method`` notation).
+
+A listener on this event can create and set a ``Response`` object, create
+and set a new ``Exception`` object, or do nothing::
+
+    use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+    use Symfony\Component\HttpFoundation\Response;
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        $exception = $event->getException();
+        $response = new Response();
+        // setup the Response object based on the caught exception
+        $event->setResponse($response);
+
+        // you can alternatively set a new Exception
+        // $exception = new \Exception('Some special exception');
+        // $event->setException($exception);
+    }
+
+.. note::
+
+    As Symfony ensures that the Response status code is set to the most
+    appropriate one depending on the exception, setting the status on the
+    response won't work. If you want to overwrite the status code (which you
+    should not without a good reason), set the ``X-Status-Code`` header::
```

Comment:
Maybe here suggest in full what are the use cases for the override or not mention it. Hint without explanation is not very helpful.